builtin/common/serialize.lua

```diff
@@ -105,12 +109,22 @@ local function serialize(value, write)
 		end
 		local type_ = type(value)
 		if type_ == "number" then
-			return write(string_format("%.17g", value))
+			if value ~= value then -- nan
+				return write"0/0"
+			elseif value == math_huge then
+				return write"1/0"
+			elseif value == -math_huge then
+				return write"-1/0"
+			else
+				return write(string_format("%.17g", value))
+			end
```

Comment:
whats this change doing here?

Reply:
@Desour wanted me to restore forward compatibility (see https://github.com/minetest/minetest/pull/12708/commits/d2af16b6f6679b75af045d1353c86c7d65ff8ed0)

Reply:
Can you split the number-related changes to a separate PR?
or at leave it as a commit and change the PR description accordingly

Reply:
Sure.